pkg/monitoring/metrics/virt-controller/vmstats_collector.go

```diff
@@ -612,3 +622,87 @@ func collectVMCreationTimestamp(vms []*k6tv1.VirtualMachine) []operatormetrics.C
 
 	return cr
 }
+
+func CollectVmsVnicInfo(vms []*k6tv1.VirtualMachine) []operatormetrics.CollectorResult {
+	var results []operatormetrics.CollectorResult
+
+	for _, vm := range vms {
+		if vm.Spec.Template == nil || vm.Spec.Template.Spec.Domain.Devices.Interfaces == nil {
+			continue
+		}
+
+		interfaces := vm.Spec.Template.Spec.Domain.Devices.Interfaces
+		networks := vm.Spec.Template.Spec.Networks
+
+		for _, iface := range interfaces {
+			bindingType, bindingName := getBinding(iface)
+			networkName, matchFound := getNetworkName(iface.Name, networks)
+
+			if !matchFound {
+				continue
+			}
+
+			results = append(results, operatormetrics.CollectorResult{
+				Metric: vmVnicInfo,
+				Labels: []string{
+					vm.Name,
+					vm.Namespace,
+					iface.Name,
+					bindingType,
+					networkName,
+					bindingName,
+				},
+				Value: 1.0,
+			})
+		}
+	}
+
+	return results
+}
+
+func getBinding(iface k6tv1.Interface) (bindingType, bindingName string) {
+	switch {
+	case iface.Masquerade != nil:
+		bindingType = "core"
+		bindingName = "masquerade"
+	case iface.Bridge != nil:
+		bindingType = "core"
+		bindingName = "bridge"
+	case iface.SRIOV != nil:
+		bindingType = "core"
+		bindingName = "sriov"
+	case iface.Binding != nil:
+		bindingType = "plugin"
+		bindingName = iface.Binding.Name
+	}
+
+	return bindingType, bindingName
+}
+
+func getNetworkName(ifaceName string, networks []k6tv1.Network) (networkName string, matchFound bool) {
+	matchFound = false
```

Comment:
The point in my original [comment](https://github.com/kubevirt/kubevirt/pull/13610#discussion_r1900384180) was to drop this ugly boolean.

In case `net` was found, return its name with `true`.
Else (no need for a real `else`), return empty string with `false`.

Reply:
+1 right sorry.
fixed